app/models/goal.py

```diff
@@ -2,4 +2,14 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship('Task', backref='goal', lazy=True)
+
+    def create_goal_dict(self):
+        goal_dict = {
+                    "id": self.goal_id,
+                    "title": self.title,
+                }
+        return goal_dict
```

Comment:
Since you immediately return this you don't need to assign it to a variable:

```suggestion
        return {
                    "id": self.goal_id,
                    "title": self.title,
                }
```